content/gocourse-2019-05-melbourne.slide

```diff
@@ -1003,6 +1003,46 @@ Use `gofmt` or `goimport` as part of your editor/IDE setup
 - ...
 
 
+* JSON - Marshal / Unmarshal
+
+- Encoding a struct or primitive converts it into a json encoded stream
+- Marshalling uses an encoder to convert a struct / primitive into an array of json bytes, which are stored in memory
+- Decoding / Unmarshalling are the opposing functions to encoding and marshalling
+
+	func Marshal(v interface{}) ([]byte, error) {
+
+	}
+
+	func Unmarshal(data []byte, v interface{}) error {
+
+	}
+
+
+* JSON - Tags
+
+- `json` tags can be attached to struct fields to control json encoding 
+
+
+	type Car struct {
+		Manufacturer string `json:"manufacturer"`
+		Model        string `json:"model"`
+		Year         int64  `json:"year,omitempty"`
+	}
+
+
+* JSON - Zero Values
+
+- Basic types will be set explicity to it's zero value, if it is omitted from the json during decoding
```

Comment:
Do you mean primitive types?
Also true for Pointers and Structs

Reply:
So I should be rephrasing this to be 
>All types will be set explicity to it's zero value, if it is omitted from the json during decoding

???

Reply:
yup, i think that's a good idea:
How about:
All Go struct fields will be set to their type's zero value if the corresponding JSON property is missing.

Reply:
done